experimenter/experimenter/segments/__init__.py

```diff
@@ -1,4 +1,5 @@
 from dataclasses import dataclass
+from typing import Dict, List, Optional
```

Comment:
Not a big deal, but in the interest of staying up-to-date with Python changes, I'd suggest forgoing these in favor of the newer python type hints:
```py
list[...]  # vs List[...]
dict[...]  # vs Dict[...]
str | None  # vs Optional[str]
```
As a bonus, don't need any imports!